### ISSUE TITLE
QR Dependencies are not working

### ISSUE THREAD

**Issue by kyledampier** [bcko/flutter_qrcode_reader#17]
I get this error and it seems to be pointing towards qr_reader, here are my logs:

```
Launching lib\main.dart on Android SDK built for x86 in debug mode...
Initializing gradle...
Resolving dependencies...
Running 'gradlew assembleDebug'...
Configuration 'compile' in project ':qrcode_reader' is deprecated. Use 'implementation' instead.
C:\Users\[user]\development\flutter\.pub-cache\hosted\pub.dartlang.org\qrcode_reader-0.3.3\android\src\main\java\com\matheusvillela\flutter\plugins\qrcodereader\QRCodeReaderPlugin.java:42: error: cannot find symbol
public class QRCodeReaderPlugin implements MethodCallHandler, ActivityResultListener, PluginRegistry.RequestPermissionResultListener {
                                                                                                    ^
  symbol:   class RequestPermissionResultListener
  location: interface PluginRegistry
C:\Users\[user]\development\flutter\.pub-cache\hosted\pub.dartlang.org\qrcode_reader-0.3.3\android\src\main\java\com\matheusvillela\flutter\plugins\qrcodereader\QRCodeReaderPlugin.java:63: error: incompatible types: QRCodeReaderPlugin cannot be converted to RequestPermissionsResultListener
            registrar.addRequestPermissionsResultListener(instance);
                                                          ^
C:\Users\[user]\development\flutter\.pub-cache\hosted\pub.dartlang.org\qrcode_reader-0.3.3\android\src\main\java\com\matheusvillela\flutter\plugins\qrcodereader\QRCodeReaderPlugin.java:155: error: method does not override or implement a method from a supertype
    @Override
    ^
Note: C:\Users\[user]\development\flutter\.pub-cache\hosted\pub.dartlang.org\qrcode_reader-0.3.3\android\src\main\java\com\matheusvillela\flutter\plugins\qrcodereader\QRCodeReaderPlugin.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
Note: Some messages have been simplified; recompile with -Xdiags:verbose to get full output
3 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':qrcode_reader:compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s
Finished with error: Gradle build failed: 1
```
Any help is appreciated!

**Comment by albertjimenez** [comment]
Same here!
`Launching lib/main.dart on Nexus 6P in debug mode...
Initializing gradle...
Resolving dependencies...
Running 'gradlew assembleDebug'...
registerResGeneratingTask is deprecated, use registerGeneratedFolders(FileCollection)
registerResGeneratingTask is deprecated, use registerGeneratedFolders(FileCollection)
registerResGeneratingTask is deprecated, use registerGeneratedFolders(FileCollection)
Configuration 'compile' in project ':qrcode_reader' is deprecated. Use 'implementation' instead.
/Users/Beruto/Documents/flutter/.pub-cache/hosted/pub.dartlang.org/qrcode_reader-0.3.3/android/src/main/java/com/matheusvillela/flutter/plugins/qrcodereader/QRCodeReaderPlugin.java:42: error: cannot find symbol
public class QRCodeReaderPlugin implements MethodCallHandler, ActivityResultListener, PluginRegistry.RequestPermissionResultListener {
                                                                                                    ^
  symbol:   class RequestPermissionResultListener
  location: interface PluginRegistry
/Users/Beruto/Documents/flutter/.pub-cache/hosted/pub.dartlang.org/qrcode_reader-0.3.3/android/src/main/java/com/matheusvillela/flutter/plugins/qrcodereader/QRCodeReaderPlugin.java:63: error: incompatible types: QRCodeReaderPlugin cannot be converted to RequestPermissionsResultListener
            registrar.addRequestPermissionsResultListener(instance);
                                                          ^
/Users/Beruto/Documents/flutter/.pub-cache/hosted/pub.dartlang.org/qrcode_reader-0.3.3/android/src/main/java/com/matheusvillela/flutter/plugins/qrcodereader/QRCodeReaderPlugin.java:155: error: method does not override or implement a method from a supertype
    @Override
    ^
Note: /Users/Beruto/Documents/flutter/.pub-cache/hosted/pub.dartlang.org/qrcode_reader-0.3.3/android/src/main/java/com/matheusvillela/flutter/plugins/qrcodereader/QRCodeReaderPlugin.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
Note: Some messages have been simplified; recompile with -Xdiags:verbose to get full output
3 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':qrcode_reader:compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

* Get more help at https://help.gradle.org

BUILD FAILED in 5s
Finished with error: Gradle build failed: 1
`

**Comment by DanChesher** [comment]
I'm also getting this same error using version 0.3.3. It looks like this problem was fixed here: #12. However, this change does not appear to have been published yet. I was able to work around this by changing my project's pubspec.yaml file to reference the most recent unpublished changes in this github repo by following the instructions here: https://flutter.io/using-packages/#dependencies-on-unpublished-packages. The updated dependency looks like this:

```
dependencies:
  ...
  qrcode_reader:
    git:
      url: git://github.com/bcko/flutter_qrcode_reader.git
```

While this is an effective temporary workaround, it would be nice if these changes could be published to a new version for this package.

**Comment by abubakarmani1** [comment]
@DanChesher Completely Agree. these changes should be published to a new version to fix issues.

**Comment by Noyal** [comment]
@DanChesher I tried your method and got this error,

FAILURE: Build failed with an exception.

* Where:
Build file 'C:\src\flutter\.pub-cache\git\flutter_qrcode_reader-d848b943a210b65df8fb58d96d63c0d33aaeee68\android\build.gradle' line: 42

* What went wrong:
A problem occurred evaluating project ':qrcode_reader'.

> Could not find method implementation() for arguments [com.dlazaro66.qrcodereaderview:qrcodereaderview:2.0.2@aar, build_d0fswkpc5vgz68eqv5kwyp9d3$_run_closure3$_closure9@1555132b] on object of type org.gradle.api.internal.artifacts.dsl.dependencies.DefaultDependencyHandler.

**Comment by DanChesher** [comment]
@Noyal I can confirm that this still works with the latest versions of everything (Flutter, Android Studio, etc.). Is your project using an older version of Gradle? That appears to be a common issue with older versions of Gradle, before the "implementation" keyword was added. More info [here](https://stackoverflow.com/q/45615474/8297786).

**Comment by Noyal** [comment]
@DanChesher I tried to run the example application and got this error. Is there any steps I should to get the app to run.

Steps I did,
1) I checked out the code from https://github.com/bcko/flutter_qrcode_reader.git ,
2) resolved the dependencies
3) ran - flutter run, in the example folder

**Comment by DanChesher** [comment]
@Noyal The example is using an older version of Gradle. I submitted a new Issue for this: #24. I also submitted a pull request to fix this. While we wait for the changes to be pulled into the master branch, take a look at the changes in the pull request to see how to change the Gradle files in your local repo to get the example running. 

Note that the example uses a local copy of the QR Code Reader plugin. This differs from the method that I mentioned previously [here](https://github.com/bcko/flutter_qrcode_reader/issues/17#issuecomment-409182690). With that method, you will always get the latest changes for this plugin from this repo. With the example method, you'll have to manually pull changes into your local clone of this repo. Both methods have advantages and disadvantages. Ideally, changes in this repo should periodically be rolled into a new published version for this plugin so that we can use pubspec.yaml to manage the version of the plugin that our projects use.

**Comment by Noyal** [comment]
@DanChesher Thanks for the new change. One more change was required for me to get the app working.

As you have suggested I changed the gradle version, but we also need to update the distributionUrl in

> example\android\gradle\wrapper\gradle-wrapper.properties

`distributionUrl=https\://services.gradle.org/distributions/gradle-4.4-all.zip`.

After this change I was able to run the application.

Thanks for the help.
